homeassistant/helpers/update_coordinator.py

```diff
@@ -138,6 +138,9 @@ async def async_refresh(self) -> None:
 
         self._debounced_refresh.async_cancel()
 
+        if self.hass.is_stopping:
```

Comment:
I wish we could do this. But after thinking about it, this use case would no longer work:

 - on shutdown, turn off X
 - if X fails to turn off, send notification

It will now always look like X failed to turn off if X is unable to fetch latest state.

Reply:
Unfortunately, I think that use case negates other any other better place to do this as well.

Anyways, thanks for taking a look.  If I come up with a solution, I'll open a new PR. 

Closing this for now.